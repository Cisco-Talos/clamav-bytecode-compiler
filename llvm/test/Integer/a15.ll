; RUN: llvm-as %s -o - | llvm-dis > %t.ll
; RUN: diff %t.ll %s.out

; test 15 bits
;
@b = constant i15 add(i15 32767, i15 1)
@c = constant i15 add(i15 32767, i15 32767)
@d = constant i15 add(i15 32760, i15 8)
@e = constant i15 sub(i15 0 , i15 1)
@f = constant i15 sub(i15 0 , i15 32767)
@g = constant i15 sub(i15 2 , i15 32767)

@h = constant i15 shl(i15 1 , i15 15)
@i = constant i15 shl(i15 1 , i15 14)
@j = constant i15 lshr(i15 32767 , i15 14)
@l = constant i15 ashr(i15 32767 , i15 14)

@n = constant i15 mul(i15 32767, i15 2)
@q = constant i15 mul(i15 -16383,i15 -3)
@r = constant i15 sdiv(i15 -1,   i15 16383)
@s = constant i15 udiv(i15 -1,   i15 16383)
@t = constant i15 srem(i15 1,    i15 32766)
@u = constant i15 urem(i15 32767,i15 -1)
@o = constant i15 trunc( i16 32768  to i15 )
@p = constant i15 trunc( i16 32767  to i15 )
@v = constant i15 srem(i15 -1,    i15 768)
 
