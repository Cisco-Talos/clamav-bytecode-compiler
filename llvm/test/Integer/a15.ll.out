; ModuleID = '<stdin>'

@b = constant i15 0                               ; <i15*> [#uses=0]
@c = constant i15 -2                              ; <i15*> [#uses=0]
@d = constant i15 0                               ; <i15*> [#uses=0]
@e = constant i15 -1                              ; <i15*> [#uses=0]
@f = constant i15 1                               ; <i15*> [#uses=0]
@g = constant i15 3                               ; <i15*> [#uses=0]
@h = constant i15 undef                           ; <i15*> [#uses=0]
@i = constant i15 -16384                          ; <i15*> [#uses=0]
@j = constant i15 1                               ; <i15*> [#uses=0]
@l = constant i15 -1                              ; <i15*> [#uses=0]
@n = constant i15 -2                              ; <i15*> [#uses=0]
@q = constant i15 16381                           ; <i15*> [#uses=0]
@r = constant i15 0                               ; <i15*> [#uses=0]
@s = constant i15 2                               ; <i15*> [#uses=0]
@t = constant i15 1                               ; <i15*> [#uses=0]
@u = constant i15 0                               ; <i15*> [#uses=0]
@o = constant i15 0                               ; <i15*> [#uses=0]
@p = constant i15 -1                              ; <i15*> [#uses=0]
@v = constant i15 -1                              ; <i15*> [#uses=0]
